src/expr/src/vector_op/agg/general_sorted_grouper.rs

```diff
@@ -69,7 +69,7 @@ impl EqGroups {
     }
 
     pub fn advance_offset(&mut self) {
-        if self.limit == 0 {
+        if (self.limit == 0) || (self.offset + self.limit >= self.indices.len()) {
```

Comment:
Any existing code that uses `EqGroups` could potentially set a limit larger than `self.len()`? Or is it just for protection against future misuse?

Reply:
It is for protection.